test/kubetest/utils.go

```diff
@@ -39,21 +38,27 @@ type NodeConf struct {
 	Dataplane *v1.Pod
 	Node      *v1.Node
 }
-type PodSupplier = func(*kube_testing.K8s, *v1.Node, string, time.Duration) *v1.Pod
-type NsePinger = func(k8s *kube_testing.K8s, from *v1.Pod) bool
-type NscChecker = func(*kube_testing.K8s, *testing.T, *v1.Pod) *NSCCheckInfo
+// PodSupplier - Type to pass supplier of pod
+type PodSupplier = func(*K8s, *v1.Node, string, time.Duration) *v1.Pod
+// NsePinger - Type to pass pinger for pod
+type NsePinger = func(k8s *K8s, from *v1.Pod) bool
+// NscChecker - Type to pass checked for pod
+type NscChecker = func(*K8s, *v1.Pod) *NSCCheckInfo
 type ipParser = func(string) (string, error)
 
-func SetupNodes(k8s *kube_testing.K8s, nodesCount int, timeout time.Duration) []*NodeConf {
+// SetupNodes - Setup NSMgr and Dataplane for particular number of nodes in cluster
+func SetupNodes(k8s *K8s, nodesCount int, timeout time.Duration) ([]*NodeConf, error) {
 	return SetupNodesConfig(k8s, nodesCount, timeout, []*pods.NSMgrPodConfig{}, k8s.GetK8sNamespace())
 }
-func SetupNodesConfig(k8s *kube_testing.K8s, nodesCount int, timeout time.Duration, conf []*pods.NSMgrPodConfig, namespace string) []*NodeConf {
+// SetupNodesConfig - Setup NSMgr and Dataplane for particular number of nodes in cluster
+func SetupNodesConfig(k8s *K8s, nodesCount int, timeout time.Duration, conf []*pods.NSMgrPodConfig, namespace string) ([]*NodeConf, error) {
 	nodes := k8s.GetNodesWait(nodesCount, timeout)
 	Expect(len(nodes) >= nodesCount).To(Equal(true),
```

Comment:
Since `SetupNodes` returns `error` probably it is better to return it at that point